motor/discovery/k8s/resolver.go

```diff
@@ -80,28 +82,45 @@ func (r *Resolver) Resolve(ctx context.Context, root *asset.Asset, tc *providers
 		return nil, err
 	}
 
-	namespace := tc.Options["namespace"]
-	if len(namespace) > 0 {
-		log.Info().Msgf("namespace filter has been set to %q", namespace)
-		namespacesFilter = namespace
+	// if --namespace and --namespaces-include were both specified, just combine them into a single
+	// list of Namespaces to allow resources from
+	namespaceOpt := tc.Options["namespace"]
+	if len(namespaceOpt) > 0 {
+		log.Info().Msgf("namespace filter has been set to %q", namespaceOpt)
+		nsFilter.include = append(nsFilter.include, namespaceOpt)
 
-		clusterNamespaces, err := p.Namespaces()
-		if err != nil {
-			return nil, err
-		}
+	}
+
+	includeNamespaces := tc.Options["namespaces-include"]
+	if len(includeNamespaces) > 0 {
+		nsFilter.include = append(nsFilter.include, strings.Split(includeNamespaces, ",")...)
+	}
+
+	// Put a Warn() message if a Namespace that doesn't exist was part of the
+	// list of Namespaces to include
+	clusterNamespaces, err := p.Namespaces()
+	if err != nil {
+		return nil, err
+	}
+	for _, ns := range nsFilter.include {
```

Comment:
I'll just repeat Chris comment on my PR, when I tried the same: https://github.com/mondoohq/cnquery/pull/256#discussion_r993473673

I tried what Chris suggested, but that made the loop more complicated. Perhaps you find an easy solution or we leave it as is for better readablility.

Reply:
Yeah, I'm not sure how to restructure it to avoid looping through the provided list to filter on...
We could use sets and do some set comparison to figure out what in the filter list doesn't exist in the actual list of namespaces, but I imagine that implementation would also have to do a similar approach.